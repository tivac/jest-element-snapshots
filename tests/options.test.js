"use strict";

const matcher = require("../index.js");

describe(".toMatchDOMSnapshot() options", () => {
    beforeEach(async () => {
        await page.goto(require.resolve("./fixtures/one.html"));
    });

    it("should accept a different page instance", async () => {
        const two = await browser.newPage();

        await two.goto(require.resolve("./fixtures/two.html"));
        
        matcher({ page : two });

        await expect(".two").toMatchDOMSnapshot();
    });

    it("should wait for an element to exist when waitFor is true", async () => {
        matcher({ waitFor : true });

        const matching = expect(".new").toMatchDOMSnapshot();

        await page.evaluate(() => {
            document.body.innerHTML = `<div class="new">NEW</div>`;
        });

        await matching;
    });
    
    it("should allow overriding the global waitFor setting", async () => {
        matcher({ waitFor : true });

        const matching = expect(".new").toMatchDOMSnapshot("snap", { waitFor : false });

        await expect(matching).rejects.toThrow("Unable to locate element matching: .new");
    });
});
