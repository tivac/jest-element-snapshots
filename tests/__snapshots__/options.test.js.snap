// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.toMatchDOMSnapshot() options should accept a different page instance 1`] = `

<div class="two">
  TWO
</div>

`;

exports[`.toMatchDOMSnapshot() options should wait for an element to exist when waitFor is true 1`] = `

<div class="new">
  NEW
</div>

`;
